core/src/main/scala/kafka/coordinator/group/CoordinatorPartitionWriter.scala

```diff
@@ -123,50 +124,79 @@ class CoordinatorPartitionWriter[T](
         val maxBatchSize = logConfig.maxMessageSize
         val currentTimeMs = time.milliseconds()
 
-        val recordsBuilder = MemoryRecords.builder(
-          ByteBuffer.allocate(math.min(16384, maxBatchSize)),
-          magic,
-          compressionType,
-          TimestampType.CREATE_TIME,
-          0L,
-          maxBatchSize
-        )
+        var recordsBuilder: MemoryRecordsBuilder = null
```

Comment:
@jeffkbkim This change cannot be applied to all cases. For all the records written by the new protocol, the atomicity matters. I think that we should revert this for now and discuss how to address this within the framework.